File: backend-crs/model/course.go
package model

import "gorm.io/gorm"

type Course struct {
	gorm.Model
	CourseCode            string             `json:"courseCode"`
	Name                  string             `json:"name"`
	TotalCapacity         int                `json:"totalCapacity"`
	CurrentlyFilled       int                `json:"currentlyFilled"`
	DepartmentID          uint               `json:"departmentId"`
	OfferedBy             Department         `json:"department" gorm:"foreignKey:DepartmentID"`
	StaffID               uint               `gorm:"uniqueIndex"`
	HandledByStaff        Staff              `json:"handledByStaff" gorm:"foreignKey:StaffID"`
	Semester              int                `json:"semester"`
	Credit                int                `json:"credit"`
	Batch                 int                `json:"batch"`
	ApplicableDepartments []CourseApplicable `gorm:"courseID"`
}

File: frontend-crs/src/model/course.ts
export interface CourseResponse {
    courses: Course[];
    status: string;
}

export interface Course {
    courseId: number;
    courseCode: string;
    name: string;
    totalCapacity: number;
    currentlyFilled: number;
    departmentId: number;
    department: Department;
    StaffID: number;
    handledByStaff: Staff;
    semester: number;
    credit: number;
    batch: number;
    applicableDepartmentIds: number[] | null;
}

export interface Department {
    id: number;
    name: string;
    code: string;
    block: string;
    ApplicableCourses: any | null;
}

export interface Staff {
    username: string;
    staffId: string;
    department: string;
    email: string;
    isClassAdvisor: boolean;
    section: string;
    batch: number;
    role: string;
}
